v3/examples/client-fasthttp/main.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2020 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+package main
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	newrelic "github.com/newrelic/go-agent/v3/newrelic"
+	"github.com/valyala/fasthttp"
+)
+
+func doRequest(txn *newrelic.Transaction) error {
+	req := fasthttp.AcquireRequest()
+	resp := fasthttp.AcquireResponse()
+	defer fasthttp.ReleaseRequest(req)
+	defer fasthttp.ReleaseResponse(resp)
+
+	req.SetRequestURI("http://localhost:8080/hello")
+	req.Header.SetMethod("GET")
+
+	ctx := &fasthttp.RequestCtx{}
+	seg := newrelic.StartExternalSegmentFastHTTP(txn, ctx)
+	defer seg.End()
+
```

Comment:
`ctx` doesn't contain any outbound req attributes. it's just an empty instance of RequestCtx.
  We would require outbound req URI and header for generating trace events.
   @mirackara

Reply:
Is this still the case if this is on the client side? In the regular [client example](https://github.com/newrelic/go-agent/blob/d0a46d00dd180231a7dd41c973c78b32bb56c70d/v3/examples/client/main.go#L15C2-L15C2) we do the same thing as well.

Reply:
@mirackara In the regular client example we are passing HTTP request object as an attribute of  **[StartExternalSegment](https://github.com/newrelic/go-agent/blob/master/v3/examples/client/main.go#L21)**  which contain outbound req parameters like URL and all.

but in the case of fastHttp you are passing fast HTTP RequestCtx object as an attribute of  [StartExternalSegmentFastHTTP](https://github.com/newrelic/go-agent/blob/master/v3/examples/client/main.go#L21)
ctx doesn't contain any outbound req attributes. it's just an empty instance of RequestCtx.


example:

```
req := fasthttp.AcquireRequest()
  .....
  ......
 req.SetRequestURI("https://www.google.com/")
 ctx := &fasthttp.RequestCtx{}
 seg := newrelic.StartExternalSegmentFastHTTP(txn, ctx)
 defer seg.End()

 err := fasthttp.Do(req, resp)

```

- Expected :
  {
  "name":"External/[www.google.com/https](http://www.google.com/https)",
  "scope":"WebTransaction/Go/GET /case1"
 }
 
 - Actual :
  {
    "name":"External/unknown/http/GET",
    "scope":"WebTransaction/Go/GET /doRequest"
 },

Reply:
Please check the following fix for outbound request

https://github.com/k2io/go-agent-fork/commit/15a72c8992aa1767f7b2596ed84d07307a1d23d3